### PR TITLE
Expand load range functionality

### PR DESCRIPTION
**Pull request by kmharrington**
Solves issue #47 . 

`load_range` will work with several different types of input date format and I added a configuration file option because that's how I want to use it.

Also wrote a lot of docs.

### CODE REVIEW

**Comment by kmharrington** [comment]
Looks like the build is failing because of `import pytz`, which I added to fix some timezone issues. I think that's a pretty standard package? It was installed by default with either python or anaconda3.

**Comment by mhasself** [comment]
I believe you need to add that package to the "requirements.txt" file.  Packages listed there are installed through pip prior to building + testing.  Since it's pip, use the pip name... which appears to be pytz.

**Comment by BrianJKoopman** [comment]
You're also missing the `pyaml` package.

**Comment by kmharrington** [comment]
We'll get this eventually! 🥇

File: python/hk/getdata.py [hunk 1]
```diff
@@ -448,7 +451,38 @@ def __init__(self, prefix, fields, start, end, index_info):
         self.index_info = index_info
 
         
-def load_range(start, stop, fields=None, alias=None, data_dir=None):
+def to_timestamp( some_time, str_format=None ):
```
Comment (mhasself) [inline python/hk/getdata.py]: No spaces around args, please... to_timestamp(some_time, ...).

File: docs/hk.rst [hunk 2]
```diff
@@ -11,6 +11,126 @@ If you're here you probably just want to read in some data, so we will
 start with that.  Later on we go into details of the data model and
 the interfaces for writing compliant HK files.
 
+Loading Data Saved OCS/SO Style
+---------------
```
Comment (mhasself) [inline docs/hk.rst]: Banner underline too short... sphinx actually complains about this.  If in emacs, try rst-mode magic:  C-c C-a C-a.

File: docs/hk.rst [hunk 3]
```diff
@@ -11,6 +11,126 @@ If you're here you probably just want to read in some data, so we will
...
+is a specific folder structure and file naming scheme. ``so3g.hk.load_range``
+is writen to load data for a specified time frame saved in that style.
+
+Example Use::
+
+    from so3g.hk import load_range    
+    data = load_range(start, stop, **kwargs)
+
+Defining Time Ranges
+```````````````````````
```
Comment (mhasself) [inline docs/hk.rst]: Line too long (and more below)!  Sphinx doesn't complain about this... but I do.  C-c C-a C-a ?

File: python/hk/getdata.py [hunk 4]
```diff
@@ -448,7 +451,38 @@ def __init__(self, prefix, fields, start, end, index_info):
...
+    Args:
+        some_time - if datetime, converted to UTC and used
+                    if int or float - assumed to be ctime, no change
+                    if string - trys to parse into datetime object
+                              - assumed to be in UTC
+        str_format - allows user to define a string format if they don't
+                    want to use a default option
+    Returns:
+        ctime of some_time
+    '''
```
Comment (mhasself) [inline python/hk/getdata.py]: This docstring (and the load_ranges one) isn't quite numpy style.  That's ok for now, we can fix it up later (when building + posting of docs is more established).

File: python/hk/getdata.py [hunk 5]
```diff
@@ -448,7 +451,38 @@ def __init__(self, prefix, fields, start, end, index_info):
...
+        str_options = ['%Y-%m-%d', '%Y-%m-%d %H:%M', '%Y-%m-%d %H:%M:%S', '%Y-%m-%d %H:%M:%S.%f']
+        for option in str_options:
+            try:
+                return to_timestamp(pytz.utc.localize( dt.datetime.strptime(some_time, option )))
+            except:
+                continue
+        hk_logger.error('Could not process string into date object, options are: {}'.format(str_options))
+        return
+    hk_logger.error('Type of date / time indication is invalid, accepts datetime, int, float, and string')
+
```
Comment (mhasself) [inline python/hk/getdata.py]: Perhaps raise ValueError on parsing failure?  I think that's what I'd expect to happen.

**Comment by kmharrington** [comment]
I might have made some git mistakes.... 

But I think I did fix all the comments 🥇

**Pull request merged**